docs/repositories-configure/using-submodules.md

```diff
@@ -4,35 +4,50 @@
 
 By default, Codacy does normal Git clones that **don't include submodules** to ensure that we only clone necessary repositories. If your organization needs to use submodules, you can request Codacy to enable this feature for you.
 
+!!! note
+    On GitHub organizations, your repository and the repositories that you add as submodules must belong to the same GitHub organization.
+
 ## Prerequisites for using submodules
 
 1.  Contact us at <mailto:support@codacy.com> asking to enable submodules on Codacy.
 
 1.  **If you're using Codacy Self-hosted**, [update your license](../chart/maintenance/license.md).
 
-1.  If your submodules are:
+1.  If your submodules are:<!--TODO PLUTO-772 Validate prerequisites-->
     -   **Public repositories**, make sure that your Git URL uses the HTTPS protocol.
     -   **Private repositories**, make sure that your Git URL uses the SSH protocol.
 
 ## Enabling submodules on a repository
 
 When using submodules, you must do the following for all your existing and new repositories:
 
-1.  Open the repository **Settings**, tab **General**. In the **Danger zone** area, you have the **SSH Key** generated by Codacy to access your repository. Take note of this key.
+1.  **For GitLab and Bitbucket**, [update the public SSH key](#update-key) that Codacy uses to access your repository.
 
-    Codacy generates this repository key when you add a repository to Codacy and uses it to clone that repository. When you're using submodules, Codacy needs to clone additional repositories it may not have access to. To overcome this, Codacy must use an SSH key of your user account to have access to the same repositories as your user.
+1.  If you're using submodules to share an analysis tool configuration file across your repositories, check if your tool recursively searches the subdirectories of your repositories for configuration files.
 
-1.  For GitHub and Bitbucket, remove this Codacy key from the repository settings on your Git provider.
+    If your tool doesn't detect the configuration files in the submodule directories, you must include a configuration file directly in the root of your repositories referencing the configuration files in the submodule directories.
 
-1.  Add a new SSH key to your git provider account by clicking the link **Add new user key** or the button **Generate New User Key**, depending on your Git provider.
+## Updating the public SSH key to access the repository {: id="update-key"}
 
-    For GitHub and Bitbucket, this takes you to the Git provider page where you can manage your user account SSH keys. For GitLab, Codacy removes the existing repository key and creates the new SSH key on your user account automatically.
+!!! info "This section applies only to GitLab and Bitbucket"
```

Comment:
If the entire section applies only to specific providers, I agree that an admonition is best 👍